syntax = "proto3";

package FILE_STORAGE;

enum FileType {
    FILE = 0;
    DIRECTORY = 1;
}

enum ContentType {
    BINARY = 0;
    TEXT = 1;
}

message ConfigPacket
{
    FileType type = 1; // directory/file
    ContentType con_type = 2; //content type -> text (1) or binary (0)
    string name = 3; // directory/file name
    uint64 chunks = 4; // number of chunks that will be sent
    uint32 chunk_size = 5; // size of each chunk
    uint64 block_size = 6; // number of symbols that will be sent
    uint32 symbol_size = 7; // size of each symbol
    uint32 overhead = 8; // number of overheads
}

//message Packet {
//    fixed32 id = 1; // ID header
//    string file_content = 2; // content from the file
//}
